### DISCUSSION TITLE
How to get started with the tileset editor for level design

### DISCUSSION THREAD

**Discussion by zFriez** [Techokami/SonicWorldsNext#44] [Q&A] [answered]
Recently I've been trying to use the tileset editor for my zone, but I'm struggling with the following things:

- Sprites for the Tilesets
- Dealing with Collisions of those Tilesets
- Slopes
- Understand the Tileset Helper

(I think it's important to note that I'm quite new to Godot and Game Dev in general. So my problem could also be not properly understanding the Godot Engine.)

I already have a new zone with a simple Green Hill platform with collision using the Tileset node, but I didn't go further than that.

My plan is to use the Green Hill spritesheet for the zone created, with slopes and collisions. How can I make that basic layout for my zone?
Is there any documentation for this topic yet?

**Comment by Renhoex** [comment]
it'd be worth reading the documentations on godot for the tilemaps
https://docs.godotengine.org/en/3.5/tutorials/2d/using_tilemaps.html

you'll want to set up the collission layer and masks to collide the way you want, they've been labeled to make it a bit clearer
![image](https://github.com/Techokami/SonicWorldsNext/assets/26295900/21957e96-bea9-4d0d-b79a-b0a6035d55bb)

we don't currently have any documentation set up since we're focusing on getting the project ported to godot 4 and anything we write for the current main build may become obsolete by the time it's finished.

**Comment by DimensionWarped** [comment, marked as answer]
I started on a brief explainer on general level design a while back, but it could use some love. If you start experimenting and have recommendations on how to improve the document let me know and I'll add them in.

https://github.com/Techokami/SonicWorldsNext/wiki/Levels#making-a-new-level

There's also a bit on collision masks in the wiki too, though none of it is specifically about tilesets.

https://github.com/Techokami/SonicWorldsNext/wiki/Collision-Layers-and-Masks

**Reply by zFriez** [reply]
Thanks. I'll play around some more with the project, as soon as I find something cool to add to the wiki, I'll be getting in touch with you.

**Comment by zFriez** [comment]
Thanks for the help, I got it working after struggling with the tilesets for quite some time 👏

Here's a little preview of it working:

![project](https://github.com/Techokami/SonicWorldsNext/assets/65882932/ba3000a4-50f1-48fe-a80c-3ca587fb22f5)
